### PR TITLE
Hotfix - Error pages

### PR DESCRIPTION
**Pull request by ArnavK-09**
Adds error handling and a custom error page.

### CODE REVIEW

**Comment by ArnavK-09** [comment]
Ahhhh wfff

**Comment by ArnavK-09** [comment]
Fixed

**Comment by ArnavK-09** [comment]
Fixed 
Fixes #386

**Comment by ArnavK-09** [comment]
> 
> 
> 

@markmead ☕🍵🧋🍷🍸🍹🍺🧉🫖🫗

File: app/[components]/[category]/[collection]/page.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import Container from '@component/Container'
 import MdxRemoteRender from '@component/MdxRemoteRender'
 import CollectionLinks from '@component/CollectionLinks'
 import CollectionList from '@component/CollectionList'
+import { notFound } from 'next/navigation'
```
Comment (markmead) [inline app/[components]/[category]/[collection]/page.jsx:13]: Please put any Next imports at the top. 

```js
import { notFound } from 'next/navigation'

// Rest
```

File: app/[components]/[category]/[collection]/page.jsx [hunk 2]
```diff
@@ -46,7 +47,7 @@ async function getCollection(params) {
     `${params.category}-${params.collection}.mdx`
   )
 
-  const postItem = await fs.readFile(componentPath, 'utf-8')
+  const postItem = await fs.readFile(componentPath, 'utf-8').catch(() => notFound())
```
Comment (markmead) [inline app/[components]/[category]/[collection]/page.jsx:50]: I'm not a fan of `.catch()`.

File: app/not-found.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { ogMeta, twitterMeta } from '@data/metadata'
+
+
+export async function generateMetadata() {
+  const description = `Uh-oh! It appears you've reached a digital crossroads, and the page you were seeking has gone missing. Don't worry, though; we're here to assist you in navigating back to our main content or helping you find the information you were looking for. Feel free to explore our site, or use the search bar to discover what you need. We apologize for any inconvenience and appreciate your patience.`
+  return {
```
Comment (markmead) [inline app/not-found.jsx:6]: This is massively overkill, it's a 404 page, not a diary entry! 😅 

Please just use this component - https://www.hyperui.dev/components/application-ui/error-pages#component-1

**Review by markmead — CHANGES_REQUESTED** [review]
Thanks for this! It's a nice idea.

**Comment by markmead** [comment]
Please improve the PR title, it's very long and over descriptive. `Hotfix - Error pages` would be fine.

**Comment by markmead** [comment]
> > 
> 
> @markmead ☕🍵🧋🍷🍸🍹🍺🧉🫖🫗

Bit cringe. Maybe don't do this on other PRs... 😅

File: app/not-found.jsx (hunk 3 above)
Comment (ArnavK-09) [reply app/not-found.jsx:6]: So remove meta tags?

**Comment by ArnavK-09** [comment]
> > > 
> > 
> > @markmead ☕🍵🧋🍷🍸🍹🍺🧉🫖🫗
> 
> Bit cringe. Maybe don't do this on other PRs... 😅 

༎ຶ⁠‿⁠༎ຶ sooo sorry, I was sleepy at that time

**Comment by ArnavK-09** [comment]
Screenshot_2023-10-08-10-06-47-60_40deb401b9ffe8e1df2f1cc5ba480b12

File: app/not-found.jsx (hunk 3 above)
Comment (ArnavK-09) [reply app/not-found.jsx]: Shorten description

File: app/[components]/[category]/[collection]/page.jsx (hunk 2 above)
Comment (ArnavK-09) [reply app/[components]/[category]/[collection]/page.jsx:50]: Ah, any other way you want Todo?

Comment (markmead) [reply app/[components]/[category]/[collection]/page.jsx:50]:
```js
if (!postItem) {
  notFound()
}
```

File: app/[components]/[category]/page.jsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { notFound } from 'next/navigation'
 import matter from 'gray-matter'
```
Comment (markmead) [inline app/[components]/[category]/page.jsx:2]: A new line would be good as these imports aren't connected.

```js
import { notFound } from 'next/navigation'

import matter from 'gray-matter'
```

File: app/not-found.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { ogMeta, twitterMeta } from '@data/metadata'
+
+
+export async function generateMetadata() {
+  const description = `Uh-oh! It appears that this page doesn't exist!`
+  return {
+    title: `404 Page Doesn't Exist | HyperUI`,
```
Comment (markmead) [inline app/not-found.jsx:7]: Just "404 | HyperUI" please.

File: app/not-found.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      title: `404 Page Doesn't Exist | HyperUI`,
+      description: description,
+      ...ogMeta,
+    },
+    twitter: {
+      title: `404 Page Doesn't Exist | HyperUI`,
+      description: description,
+      ...twitterMeta,
+    },
+  }
```
Comment (markmead) [inline app/not-found.jsx:19]: No need for backticks.

File: app/not-found.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export default function Custom404() {
+  return (
+    <div className="grid h-screen px-4 bg-white place-content-center">
+      <h1 className="tracking-widest text-gray-500 text-center uppercase text-3xl">404 | URL Not Found!</h1>
```
Comment (markmead) [inline app/not-found.jsx:25]: This isn't how the component I linked looks.

**Review by markmead — CHANGES_REQUESTED** [review]

File: app/not-found.jsx (hunk 7 above)
Comment (ArnavK-09) [reply app/not-found.jsx]: > https://www.hyperui.dev/components/application-ui/error-pages#component-1

You linked to components#1 (simple), I just added  ` text-center `, now it's the same to same component you linked, sorry

**Comment by ArnavK-09** [comment]
> @markmead 



Already tried that, **if I used if else, it would throw an error**, resulting in 500 in production & error log in development!

(Unable to reply in that review message)

File: app/[components]/[category]/[collection]/page.jsx (hunk 2 above)
Comment (ArnavK-09) [reply app/[components]/[category]/[collection]/page.jsx:50]: Already tried that, **it throws error, if I would done this**, results in 500 in production, error log in development

**Comment by markmead** [comment]
OK, I'll pull it down and have a look.

**Comment by markmead** [comment]
Pushed some updates and clean up the PR comments. Take a look and see what you think.

**Comment by ArnavK-09** [comment]
Status?

**Comment by ArnavK-09** [comment]
> Pushed some updates and clean up the PR comments. Take a look and see what you think.

Looks good

**Review by markmead — APPROVED** [review]
LGTM!

**Pull request merged**